### ISSUE TITLE
ImportError when using default credentials from Managed VMs with non-default runtime

### ISSUE THREAD

**Issue by jonparrott** [google/oauth2client#184]
When using app engine managed VMs without using the "python-runtime" base, (eg, to use python3 instead of 2.7) getting default credentials breaks because "google.appengine" (the app engine sdk) does not exist. ``_get_application_default_credential_GAE`` should treat the absence of the SDK as indication that we are not running on normal GAE and to try other options.

https://github.com/google/oauth2client/blob/master/oauth2client/client.py#L1082

Stacktrace:
```
Traceback (most recent call last):
  File "main.py", line 1, in <module>
    from bookshelf import app
  File "/app/bookshelf/__init__.py", line 2, in <module>
    from .crud import crud
  File "/app/bookshelf/crud.py", line 2, in <module>
    from . import model_datastore as model
  File "/app/bookshelf/model_datastore.py", line 5, in <module>
    datastore.set_defaults(dataset_id='durable-pipe-855')
  File "/env/src/gcloud/gcloud/datastore/__init__.py", line 89, in set_defaults
    set_default_connection(connection=connection)
  File "/env/src/gcloud/gcloud/datastore/_implicit_environ.py", line 133, in set_default_connection
    connection = connection or get_connection()
  File "/env/src/gcloud/gcloud/datastore/_implicit_environ.py", line 124, in get_connection
    return Connection.from_environment()
  File "/env/src/gcloud/gcloud/connection.py", line 174, in from_environment
    credentials = get_credentials()
  File "/env/src/gcloud/gcloud/credentials.py", line 73, in get_credentials
    return client.GoogleCredentials.get_application_default()
  File "/env/lib/python3.4/site-packages/oauth2client/client.py", line 1199, in get_application_default
    return GoogleCredentials._get_implicit_credentials()
  File "/env/lib/python3.4/site-packages/oauth2client/client.py", line 1184, in _get_implicit_credentials
    credentials = checker(env_name=env_name)
  File "/env/lib/python3.4/site-packages/oauth2client/client.py", line 1099, in _implicit_credentials_from_gae
    return _get_application_default_credential_GAE()
  File "/env/lib/python3.4/site-packages/oauth2client/client.py", line 1369, in _get_application_default_credential_GAE
    from oauth2client.appengine import AppAssertionCredentials
  File "/env/lib/python3.4/site-packages/oauth2client/appengine.py", line 31, in <module>
    from google.appengine.api import app_identity
ImportError: No module named 'google.appengine'
```

**Comment by dhermes** [comment]
Should this be closed by #185?

**Comment by jonparrott** [comment]
Yep, beat me to it. Closed by #185.
